sessionctx/variable/session.go

```diff
@@ -2047,16 +2083,14 @@ func (s *SessionVars) GetTemporaryTable(tblInfo *model.TableInfo) tableutil.Temp
 // EncodeSessionStates saves session states into SessionStates.
 func (s *SessionVars) EncodeSessionStates(ctx context.Context, sessionStates *sessionstates.SessionStates) (err error) {
 	// Encode user-defined variables.
-	s.UsersLock.RLock()
```

Comment:
Should lock it here since the following operation is accessing the user-var map directly instead of using `Getter/Setter`.